### DISCUSSION TITLE
Expect a file object (PDF or Excel) and Open It

### DISCUSSION THREAD

**Discussion by HGamalElDiin** [fastapi/fastapi#8566] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@App.post("/Recognize")
async def Recognize(FILE: UploadFile = File('FILE')):
    OBJECT = main(FILE)
    return OBJECT

def FILE_READER(FILE):
    before = timer()
    try:
        suffix = Path(FILE.filename).suffix
        print(suffix)
        with NamedTemporaryFile(delete=False, suffix=suffix) as tmp:
            shutil.copyfileobj(FILE.file, tmp)
            PATH = Path(tmp.name)
        CONTENT = ""
        if suffix == ".pdf":
            doc = fitz.open(PATH)
            for page in doc:
                CONTENT += page.getText()
        elif suffix == ".xlsx":
            CONTENT = pd.read_excel(PATH, engine='openpyxl')
        elif suffix ==  ".csv":
            CONTENT = pd.read_csv(PATH)
    finally:
        FILE.file.close()
        tmp.close()
        print(tmp.closed)
        print(PATH)
        #os.remove(PATH)
    after = timer()
    print("Full Process time: {}".format(timedelta(seconds=after-before)))
    return CONTENT
```

### Description

I read the documentation and have searched a lot on how to expect a file comes to the FastAPI (not bytes) from the .NET C# backend to FastAPI python app after user uploading it, and open it after checking its extension to use the appropriate file reader, i can't get any useful tutorials and it's not working with file.read() or file.file.read() i would say.

What I'm doing now, is getting the file, write it in a temporary location, then check it's suffix and open it.. can't be done without writing the file first.

### Operating System

Windows

### Operating System Details

Windows 10 pro

### FastAPI Version

0.70.0

### Python Version

3.7

**Comment by blokje** [comment]
This is related to a https://github.com/encode/starlette/issues/697. Starlette does not offer a method to override the tempfile handler at this moment and as a result neither does FastAPI.  But maybe I'm missing something here, @tiangolo?

So right now, you are handling it "correctly" although it feel a bit odd.

See also: https://github.com/tiangolo/fastapi/issues/1838 and https://github.com/tiangolo/fastapi/issues/2512
